### ISSUE TITLE
Java 14 SwitchExpression Check Support validation: DefaultComesLastCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8665]
Child of #8658 
Check documentation: https://checkstyle.sourceforge.io/config_coding.html#DefaultComesLast

_Check that the default is after all the cases in a switch statement._

```
➜  src /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java     
➜  src cat config.xml                                                                                                   
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="DefaultComesLast"/>
    </module>
</module>
➜  src cat TestClass.java                                                                                               
public class TestClass {
    public int method1(int i) {
        int x = 7;
        switch (i) {
            case 1:
            default: // violation, 'default' before 'case'
                x=9;
                break;
            case 2:
                x=8;
                break;
        }
        return x;
    }

    public int method2(int i) {
        int x = 7;
        switch (i) {
            case 1 -> {
                x=8;
            }
            default ->{ // should be violation, 'default' before 'case'
                x = 9;
            }
            case 2 -> {
                x = 7;
            }
        }

        return x;
    }

    public int method3(int i) {
        return switch (i) {
            case 1 -> 8;
            default -> 9; // should be violation, 'default' before 'case'
            case 2 -> 7;
        };
    }
}


➜  src java -jar /home/nick/IdeaProjects/checkstyle/target/checkstyle-8.36-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/enhanced-switch/src/TestClass.java:6:13: Default should be last label in the switch. [DefaultComesLast]
Audit done.
Checkstyle ends with 1 errors.

```
We need to update this check to support switch rules and expressions.

**Comment by pbludov** [comment]
I am surprised that there is no compilation error here. There will be issues once the patten matching come to the `switch` expression:
```java
    switch (obj) {
        case "X" -> "x literal"
        case String s  -> "some string"
        default -> "something else"
    };
```
For this example the order of `case`s is important. And `default` definitely should be the last.

**Comment by pbludov** [comment]
Fix is merged.
